### DISCUSSION TITLE
Question on maximum basket size

### DISCUSSION THREAD

**Discussion by tpalgut** [imRohan/Pantry#116] [Q&A] [answered]
I'm wondering what the largest size data that I can place in a basket. I'm getting a status code 413 with a 'Payload Too Large' message. In this particular instance the data size is 123095 bytes.

**Comment by imRohan** [comment, marked as answer]
@tpalgut That's a good question!

There are 2 size limitations that have been added to Pantry.

First, there is a hard limit of 1.44mb for each basket ([link to line](https://github.com/imRohan/Pantry/blob/efa5ad063518c4a97f4d12992da97e1c2e6607ce/src/decorators/block.ts#L12)), and second, it seems like the middleware I'm using to parse the request (body-parser) has a default limit of 100kb per transaction.

I'll create an issue to fix the 100kb request size (which will solve your 413 response issue), however, you will still not be able to have a basket with a max size greater than 1.44mb.

Depending on performance and costs, I could bump the max size to 2mb.

**Reply by tpalgut** [reply]
Hello @imRohan
Thank you for the response and support.  I think that I am hitting the 100kb limit.  It would be great if that could get increased.  The 1.44 mb limit is good for me.  I won't need more.
A little background on this data.  I am a 'birder' and I have some scripts that scrape rare bird sighting data out of emails and place it in the basket.  The basket data can then be used by web sites or mobile apps to map the locations for other birders to reference.  It's a work in progress.

**Reply by imRohan** [reply]
Hi @tpalgut, I've submitted a PR and will be updating Pantry to support a maximum of 1mb per request.

Your project sounds really interesting! I have a friend who's also a birder so please keep me posted - i'd love to pass on any project details for them to also enjoy.

Cheers,

-R
